File: Laboratorios/Lab2/Fibo/fibo.c
#include<stdio.h>

short fiboS(short n){
    short a = 0, b = 1, f = 0,i;
    if(n<2)return n;
    for(i = 1; i<n; i++){
        f = a+b;
        a = b;
        b = f;
    }
    return f;
}

int fiboI(int n){
    int a = 0, b = 1, f = 0,i;
    if(n<2)return n;
    for(i = 1; i<n; i++){
        f = a+b;
        a = b;
        b = f;
    }
    return f;
}

float fiboF(float n){
    float a = 0, b = 1, f = 0,i;
    if(n<2)return n;
    for(i = 1; i<n; i++){
        f = a+b;
        a = b;
        b = f;
    }
    return f;
}

long fiboL(long n){
    long a = 0, b = 1, f = 0, i;
    if(n<2)return n;
    for(i = 1; i<n; i++){
        f = a+b;
        a = b;
        b = f;
    }
    return f;
}

long long fiboLl(long long n){
    long long a = 0, b = 1, f = 0,i;
    if(n<2)return n;
    for(i = 1; i<n; i++){
        f = a+b;
        a = b;
        b = f;
    }
    return f;
}

double fiboD(double n){
    double a = 0, b = 1, f = 0,i;
    if(n<2)return n;
    for(i = 1; i<n; i++){
        f = a+b;
        a = b;
        b = f;
    }
    return f;
}

long double fiboLD(long double n){
    long double a = 0, b = 1, f = 0,i;
    if(n<2)return n;
    for(i = 1; i<n; i++){
        f = a+b;
        a = b;
        b = f;
    }
    return f;
}

int main(){
    printf("n: %i, fiboShort: %i\n", 24, fiboS(24));
    printf("n: %i, fiboInt: %i\n", 47, fiboI(47));
    printf("n: %i, fibo: %f\n", 37, fiboF((float)37));
    printf("n: %d, fibo: %d\n", 47, fiboL(47));
    printf("n: %d, fibo: %d\n", 47, fiboLl(47));
    printf("n: %d, fibo: %i\n", 50, fiboD(46));
    printf("n: %L, fibo: %L\n", 50, fiboLD(46));
}

File: Laboratorios/Lab2/Fibo/fibo.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 *
 * @author origa
 */


public class fibo {
    
    static short fiboS(short n){
        short a = 0, b = 1, f = 0,i;
        if(n<2)return n;
        for(i = 1; i<n; i++){
            f = (short) (a+b);
            a = b;
            b = f;
        }
        return f;
    }

    static int fiboI(int n){
        int a = 0, b = 1, f = 0,i;
        if(n<2)return n;
        for(i = 1; i<n; i++){
            f = a+b;
            a = b;
            b = f;
        }
        return f;
    }

    static float fiboF(float n){
        float a = 0, b = 1, f = 0,i;
        if(n<2)return n;
        for(i = 1; i<n; i++){
            f = a+b;
            a = b;
            b = f;
        }
        return f;
    }

    static long fiboL(long n){
        long a = 0, b = 1, f = 0, i;
        if(n<2)return n;
        for(i = 1; i<n; i++){
            f = a+b;
            a = b;
            b = f;
        }
        return f;
    }


    static double fiboD(double n){
        double a = 0, b = 1, f = 0,i;
        if(n<2)return n;
        for(i = 1; i<n; i++){
            f = a+b;
            a = b;
            b = f;
        }
        return f;
    }

    public static void main(String[] args) {
        System.out.println("FiboShort(" + 23 + ") "+ fiboS((short)24));
        System.out.println("FiboInt(" + 46 + ") "+ fiboI((int)47));
        System.out.println("FiboFloat(" + 37 + ") "+ fiboF((float)37));
        System.out.println("FiboLong(" + 93 + ") "+ fiboL((long)93));
        System.out.println("FiboDouble(" + 1477 + ") "+ fiboD((double)1477));
    }
    
}
